### PR TITLE
Fix mypy errors - part 4

### PR DESCRIPTION
**Pull request by e-larsson**
Part of #51

### CODE REVIEW

File: scanomatic/image_analysis/image_grayscale.py [hunk 1]
```diff
@@ -422,8 +432,9 @@ def detect_grayscale(im_trimmed, grayscale: Grayscale):
                     np.save(os.path.join(
                         Paths().log,
                         f"gs_segment_{i}.npy",
+                    ),
```
Comment (skymandr) [inline scanomatic/image_analysis/image_grayscale.py:435]: Strange indentation, better to break before `os.path` and indent accordingly?

Comment (BengtRydberg) [inline scanomatic/image_analysis/image_grayscale.py:435]: den här radbrytningen ser lite knepig ut.

File: scanomatic/image_analysis/image_grayscale.py [hunk 2]
```diff
@@ -437,9 +448,10 @@ def detect_grayscale(im_trimmed, grayscale: Grayscale):
             ),
         )
 
-    if len(grayscale_segment_centers) == 0:
+    if grayscale_segment_centers is None:
```
Comment (local-minimum) [inline scanomatic/image_analysis/image_grayscale.py:451]: they can't have length 0 still?

File: tools/signal.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+
+        for i, s in enumerate([s - int(s) for s in shifts]):
+            if s != 0:
+                signals[i][:] = np.array(
+                    [sign + s * frequencies[i] for sign in signals[i]]
+                )
+        return signals
+
+
+def get_continious_slopes(
```
Comment (skymandr) [inline tools/signal.py:69]:
```suggestion
def get_continuous_slopes(
```

**Review by local-minimum — COMMENTED** [review]
generally looks good, only one question

File: tools/signal.py [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+    new_list = []
+    for i in ideal_signal:
+        delta_i = np.abs(X - i)
+        delta_reciprocal = np.abs(Y - X[delta_i.argmin()])
+        if delta_i.min() == delta_reciprocal.min():
+            new_list += ([X[delta_i.argmin()],
+                Y[delta_reciprocal.argmin()]])
+
+    return np.array(new_list)
+"""
```
Comment (skymandr) [inline tools/signal.py:178]: Remove and rename `_get_alt_closest` to `_get_closest`?

File: tools/signal.py [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+    ].ravel()
+    return Z
+
+
+def _get_orphans(X: np.ndarray, shortX: np.ndarray) -> np.ndarray:
+    return X[np.abs(np.subtract.outer(X, shortX)).min(axis=1).astype(bool)]
+
+
+def get_offset_quality(
+    s,
```
Comment (skymandr) [inline tools/signal.py:197]: This and `w1` aren't very nice. Maybe @local-minimum has ideas for better names? (Here and below.)

**Review by skymandr — COMMENTED** [review]
Not much to add.

File: tools/signal.py (hunk 5 above)
Comment (local-minimum) [reply tools/signal.py:197]: nope and the point of `/tools` is to not really maintain the code until we notice we or someone needs it so I would argue to put minimal time fixing things here... perhaps typos but not more.

Comment (skymandr) [reply tools/signal.py:197]: Is the plan in the long run, then, to remove this directory?

File: tools/signal.py (hunk 4 above)
Comment (e-larsson) [reply tools/signal.py]: Fixed

File: scanomatic/image_analysis/image_grayscale.py (hunk 1 above)
Comment (e-larsson) [reply scanomatic/image_analysis/image_grayscale.py]: Fixed

File: scanomatic/image_analysis/image_grayscale.py (hunk 2 above)
Comment (e-larsson) [reply scanomatic/image_analysis/image_grayscale.py]: I guess that could be possible still. Added it again

File: scanomatic/image_analysis/image_grayscale.py (hunk 1 above)
Comment (e-larsson) [reply scanomatic/image_analysis/image_grayscale.py]: Fixat

File: tools/signal.py (hunk 5 above)
Comment (local-minimum) [reply tools/signal.py:182]: move it to a new repo

**Review by local-minimum — APPROVED** [review]

**Pull request merged**